### PR TITLE
Add blacklist file types.

### PR DESCRIPTION
**Pull request by amine3**

### CODE REVIEW

**Review by elyousfi5 — APPROVED** [review]
lgtm

**Review by PiranhaSa — APPROVED** [review]

File: agent/osv_agent.py [hunk 1]
```diff
@@ -153,9 +193,16 @@ def _emit_vulnerabilities(
     def _process_asset_file(self, message: m.Message) -> None:
         """Process message of type v3.asset.file."""
         content = _get_content(message)
+        path = message.data.get("path")
         if content is None or content == b"":
             logger.warning("Message file content is empty.")
             return
+        file_type = _get_file_type(content, path)
+        logger.info("Analyzing file `%s` with type `%s`.", path, file_type)
```
Comment (deadly-panda) [inline agent/osv_agent.py:201]: `debug` .. would be noisy no?

File: agent/osv_agent.py [hunk 2]
```diff
@@ -113,6 +141,18 @@ def _run_osv(path: str, content: bytes) -> str | None:
...
 
 
+def _get_file_type(content: bytes, path: str | None) -> str:
+    if path is None:
+        mime = magic.from_buffer(content, mime=True)
+        file_type = mimetypes.guess_extension(mime)
+        return str(file_type)
+    else:
+        file_split = os.path.splitext(path)[1]
+        if len(file_split) < 2:
```
Comment (mohsinenar) [inline agent/osv_agent.py:151]: Why 2? Please add a comment or move it to a const so its clear!

**Review by mohsinenar — CHANGES_REQUESTED** [review]

File: agent/osv_agent.py [hunk 3]
```diff
@@ -48,6 +51,31 @@
...
+    ".psd",
+    ".so",
+    ".strings",
+    ".svg",
+    ".symbols",
+    ".ttf",
+    ".woff",
+    ".woff2",
+    ".zip",
+    ".xml",
```
Comment (3asm) [inline agent/osv_agent.py:77]: xml no, they have lock files in XML formats.

Comment (3asm) [inline agent/osv_agent.py:77]: xml no, some lock files are XML. You need to double check the sbom and lock files list please.

Comment (3asm) [reply agent/osv_agent.py:77]: I would even suggest whitelisting the support formats instead.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**